test/fork.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {expect} from 'chai';
+import {describe, it, before} from 'mocha';
+import {logger, octokit, setup} from './util';
+import * as sinon from 'sinon';
+import {fork} from '../src/github-handler/fork-handler';
+
+before(() => {
+  setup();
+});
+
+describe('Fork', async () => {
+  const upstreamOwner = 'upstream-owner';
+  const upstreamRepo = 'upstream-repo';
+  const responseData = await import('./fixtures/create-fork-response.json');
+
+  describe('Octokit createFork', () => {
+    it('Is called with the correct values', () => {
+      const createRefResponse = {
+        headers: {},
+        status: 200,
+        url: 'http://fake-url.com',
+        data: responseData,
+      };
+      // setup
+      const stub = sinon
+        .stub(octokit.repos, 'createFork')
+        .resolves(createRefResponse);
+      // tests
+      fork(logger, octokit, upstreamOwner, upstreamRepo);
+      sinon.assert.calledOnceWithExactly(stub, {
+        owner: upstreamOwner,
+        repo: upstreamRepo,
+      });
+      // restore
+      stub.restore();
+    });
+  });
+
+  describe('Forking function', () => {
+    it('Returns correct values on success', () => {
+      const createRefResponse = {
+        headers: {},
+        status: 200,
+        url: 'http://fake-url.com',
+        data: responseData,
+      };
+      // setup
+      const stub = sinon
+        .stub(octokit.repos, 'createFork')
+        .resolves(createRefResponse);
+      // tests
+      fork(logger, octokit, upstreamOwner, upstreamRepo);
+      sinon.assert.calledOnceWithExactly(stub, {
+        owner: upstreamOwner,
+        repo: upstreamRepo,
+      });
+      // restore
+      stub.restore();
+    });
+    it('Passes the error message with a throw when octokit fails', async () => {
+      // setup
+      const errorMsg = 'Error message';
+      const stub = sinon.stub(octokit.repos, 'createFork').rejects(errorMsg);
+      try {
+        await fork(logger, octokit, upstreamOwner, upstreamRepo);
+        throw Error('Test should not have passed');
```

Comment:
There's probably a better `fail()` method that can be used rather than throwing an error. @bcoe Any suggestions?

Reply:
Ah yes I found one. `expect.fail();`
https://stackoverflow.com/a/55455874